clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1011,39 +1009,6 @@ protected void produceToStoreBufferServiceOrKafka(
 
       hostLevelIngestionStats.recordStorageQuotaUsed(storageUtilizationManager.getDiskQuotaUsage());
     }
-
-    if (whetherToApplyThrottling) {
-      /**
-       * We would like to throttle the ingestion by batches of ({@link PubSubMessage} returned by each poll.
-       * The batch shouldn't be too big, otherwise, the {@link StoreBufferService#putConsumerRecord(PubSubMessage, StoreIngestionTask, LeaderProducedRecordContext, int, String, long)}
-       * could be blocked when the buffer is full, and the throttling could be inaccurate.
-       * So every record returned from {@link com.linkedin.venice.pubsub.consumer.PubSubConsumer#poll(long)} should be processed
-       * as fast as possible to avoid long-lasting objects in JVM to minimize the 'object copy' time
-       * during GC.
-       *
-       * Here are more details:
-       * 1. Previously, the throttling was happening in StoreIngestionTask#processConsumerRecord,
-       *    which would be invoked by StoreBufferService;
-       * 2. When the ingestion got throttled, the database operations would halt, but the deserialized
-       *    records would be kept pushing to the intermediate buffer pool until the pool is full;
-       * 3. When Young GC happens, all the objects in the buffer pool could be potentially copied to Survivor
-       *    space since they are being actively referenced;
-       * 4. The object copy time is the slowest phase in Young GC;
-       *
-       * By moving the throttling logic after putting deserialized records to buffer, the records in the buffer
-       * will be processed as long as there is enough capacity.
-       * With this way, the actively referenced object will be reduced greatly during throttling and Young GC
-       * won't need to copy many objects from Young regions to Survivor regions, which reduces the overall
-       * GC pause time.
-       */
-      bandwidthThrottler.maybeThrottle(totalBytesRead);
-      recordsThrottler.maybeThrottle(recordCount);
-
-      if (!orderedWritesOnly) {
-        unorderedBandwidthThrottler.maybeThrottle(totalBytesRead);
-        unorderedRecordsThrottler.maybeThrottle(recordCount);
```

Comment:
Thanks for cleaning this up! I believe these 6 class properties (4 throttlers, `orderedWritesOnly`, and `recordCount`) are all unused after deleting them from here. Can you check? If that's true, then let's get rid of them completely. Let's also remove the related getters and setters from `StoreIngestionTaskFactory.Builder`. All of that is now handled by the `ConsumptionTask`. It's good to do a spring cleaning :D !